python/dgl/graph_index.py

```diff
@@ -51,21 +51,19 @@ def __setstate__(self, state):
         self._multigraph = multigraph
         self._readonly = readonly
         if readonly:
-            self._init(src, dst, utils.toindex(F.arange(0, len(src))), n_nodes)
+            self._init(src, dst, n_nodes)
         else:
             self._handle = _CAPI_DGLGraphCreateMutable(multigraph)
             self.clear()
             self.add_nodes(n_nodes)
             self.add_edges(src, dst)
 
-    def _init(self, src_ids, dst_ids, edge_ids, num_nodes):
+    def _init(self, src_ids, dst_ids, num_nodes):
```

Comment:
is it possible that users may provide their own edge ids?

Reply:
I don't see any usage in DGL. For example, when converting nx graph to DGL graph, we will first figure out the right edge id order and pack the edge features into one tensor by that order so they are consistent.